pandapower/plotting/patch_makers.py

```diff
@@ -257,29 +300,112 @@ def sgen_patches(node_coords, size, angles, **kwargs):
     polys, lines = list(), list()
     offset = kwargs.get("offset", 2 * size)
     r_triangle = kwargs.get("r_triangles", size * 0.4)
+    hub_size = size * 0.15  # Hub size
+    blade_coord1 = size * 0.08    # Blade angle coord
+    blade_coord2 = size * 0.22
+    pv_rect_size = size*2
+    pv_tri_size = size
     edgecolor = kwargs.get("patch_edgecolor", "w")
     facecolor = kwargs.get("patch_facecolor", "w")
     edgecolors = get_color_list(edgecolor, len(node_coords))
     facecolors = get_color_list(facecolor, len(node_coords))
+    Path = mpath.Path
+
+    if unique_angles is None:
+        unique_angles = {}
+
     for i, node_geo in enumerate(node_coords):
-        mid_circ = node_geo + _rotate_dim2(np.array([0, offset + size]), angles[i])
-        circ_edge = node_geo + _rotate_dim2(np.array([0, offset]), angles[i])
-        mid_tri1 = mid_circ + _rotate_dim2(np.array([r_triangle, -r_triangle / 4]), angles[i])
-        mid_tri2 = mid_circ + _rotate_dim2(np.array([-r_triangle, r_triangle / 4]), angles[i])
-        # dropped perpendicular foot of triangle1
-        perp_foot1 = mid_tri1 + _rotate_dim2(np.array([0, -r_triangle / 2]), angles[i])
-        line_end1 = perp_foot1 + + _rotate_dim2(np.array([-2.5 * r_triangle, 0]), angles[i])
-        perp_foot2 = mid_tri2 + _rotate_dim2(np.array([0, r_triangle / 2]), angles[i])
-        line_end2 = perp_foot2 + + _rotate_dim2(np.array([2.5 * r_triangle, 0]), angles[i])
-        polys.append(Circle(mid_circ, size, fc=facecolors[i], ec=edgecolors[i]))
-        polys.append(RegularPolygon(mid_tri1, numVertices=3, radius=r_triangle,
-                                    orientation=-angles[i], fc=facecolors[i], ec=edgecolors[i]))
-        polys.append(RegularPolygon(mid_tri2, numVertices=3, radius=r_triangle,
-                                    orientation=np.pi - angles[i], fc=facecolors[i],
-                                    ec=edgecolors[i]))
-        lines.append((node_geo, circ_edge))
-        lines.append((perp_foot1, line_end1))
-        lines.append((perp_foot2, line_end2))
+        if len(unique_angles) != 0 and "WT" in unique_angles[i]["sgen"]:
+            try:
+                angle = unique_angles[i]["sgen"]["WT"] + (angles[i] or 0)
+            except IndexError:
+                continue
```

Comment:
What would an index error mean? In what case could it occur? I wouldn't necessarily "hide" an exception that is thrown, because it could indicate a wrong implementation somewhere, except we are sure that it is more useful to just not plot some node elements when the angles or unique_angles are not given correctly.

Reply:
There were some edge cases where angles were not in sync with node_coords so used exceptions, This is removed in the upcoming commit